code/modules/atmospherics/machinery/components/unary_devices/vent_scrubber.dm

```diff
@@ -238,6 +238,9 @@
 	if("toggle_filter" in signal.data)
 		filter_types ^= gas_id2path(signal.data["toggle_filter"])
 
+	if("set_filters" in signal.data)
+		filter_types = signal.data["set_filters"]
```

Comment:
hey sorry, I wasn't quite clear enough. do this instead:
```
filter_types = list()
for(var/gas in signal.data["set_filters"])
    filter_types += gas_id2path(gas)
```
the conversion proc already handles receiving a path rather than an ID so don't worry there.